### DISCUSSION TITLE
Custom turn order that automatically skips turns

### DISCUSSION THREAD

**Discussion by dondragon2** [boardgameio/boardgame.io#904] [Q&A] [answered]
I am build a card game where each player has 1 move per turn, however, if a player does not fit particular criteria and it is their turn then the game should automatically pass them and give the turn to the next player in line.

I have tried to evaluate the skip criteria in the onBegin of turn but this does not work.

What is the best way to implement this type of feature?

**Comment by delucis** [comment, marked as answer]
@dondragon2 You can implement this using [a custom turn order](https://boardgame.io/documentation/#/turn-order?id=creating-a-custom-turn-order). Here’s a basic example that re-implements the default behaviour:

```js
const game = {
  turn: {
    order: {
      // Get the initial value of playOrderPos at the beginning of the phase.
      first: (G, ctx) => 0,

      // Get the next value of playOrderPos at the end of each turn.
      next: (G, ctx) => {
        const nextPos = (ctx.playOrderPos + 1) % ctx.numPlayers;
        return nextPos;
      },
    },
  },
};
```

In your case, you would find the next player based on your criteria and return their position in the play order. For example, if you have some function that can return the next player ID, you might write a next method like:

```js
next: (G, ctx) => {
  const playerID = getNextPlayerID(G, ctx);
  const playOrderPos = ctx.playOrder.indexOf(playerID);
  return playOrderPos;
},
```

(To be clear, `playOrderPos` is different from `playerID` because you might have a `playOrder` of `['1', '2', '0']`, in which `playerID` `'1'` has `playOrderPos` `0`, but if you never set `playOrder` to a custom order `playerID`s will just be string versions of their `playOrderPos`.)

---

Here’s a demo of a game where players can be eliminated that uses some similar ideas to skip eliminated players: https://codesandbox.io/s/boardgameio-elimination-demo-2rezv?file=/src/Game.js

**Comment by dondragon2** [comment]
thanks @delucis that is working as expected.
